test/parallel/test-http-agent-error-on-idle.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+var common = require('../common');
+var assert = require('assert');
+var http = require('http');
+var Agent = http.Agent;
+
+var agent = new Agent({
+  keepAlive: true,
+});
+
+function get(path, callback) {
+  return http.get({
+    host: 'localhost',
+    port: common.PORT,
+    agent: agent,
+    path: path
+  }, callback);
+}
+
+var destroy_queue = {};
+var server = http.createServer(function(req, res) {
+  res.end('hello world');
+});
+
+server.listen(common.PORT, function() {
+  get('/', function(res) {
+    assert.equal(res.statusCode, 200);
+    res.resume();
+    res.on('end', function() {
+      process.nextTick(function () {
+        var freeSocket = agent.freeSockets[Object.keys(agent.freeSockets)[0]][0];
```

Comment:
Let's check `freeSockets` with `assert` before this line. We want to be sure that there is a free socket after all!